corehq/apps/user_importer/importer.py

```diff
@@ -389,11 +391,37 @@ def create_or_update_users_and_groups(domain, user_specs, group_memoizer=None, u
                         user.reset_locations(location_ids, commit=False)
 
                 if role:
-                    user.set_role(domain, roles_by_name[role].get_qualified_id())
+                    role = roles_by_name[role].get_qualified_id()
+                    user.set_role(domain, role)
+
+                if web_user:
+                    user.user_data.update({'login_as_user': web_user})
 
                 user.save()
 
-                if send_account_confirmation_email:
+                if web_user:
+                    current_user = CouchUser.get_by_username(web_user)
+                    if not current_user and is_account_confirmed:
+                        raise UserUploadError(_(
+                            f"You can only set 'Is Account Confirmed' to 'True' on an existing Web User. {web_user} is a new username."
+                        ))
+                    if current_user and not current_user.is_member_of(domain) and is_account_confirmed:
+                        current_user.add_as_web_user(domain, role=role, location_id=user.location_id)
```

Comment:
I think this is a change to existing workflows, yeah? Like we don't allow directly adding web users to a domain except now through this mechanism, right? I assume that is desired and product implications are thought through? (I can't imagine a way to abuse adding someone to a domain other than hurting their commcare UX, but just thought it was worth stating)

Reply:
This is definitely a change, but hopefully a good one, and one that was requested, especially since these users will be in many domains, and making them click through a ton of invites is not a great UX. We can already kind of do this with a management command now that accepts invites for a user, so it would not surprise me if these users are used to that workflow anyway.

I admittedly hadn't thought about people abusing that feature in that way, but yeah, the only harm seems to be making someone's domain list longer, which seems minor at best